packages/next-source-maps/index.js

```diff
@@ -1,4 +1,6 @@
-module.exports = (nextConfig = {}) => {
+module.exports = (nextConfig = {}, options = {}) => {
```

Comment:
Let's change this to `module.exports = (options) => (nextConfig) => {`

Usage would be:

```js
const withSourceMaps = require('@zeit/next-source-maps')({
  devtool: 'hidden-source-map'
})

module.exports = withSourceMaps(
  {
    webpack(config, options) {
      return config
    }
  }
)
```

Reply:
I know this would be a breaking change, but we'd just release a major version. I'd like all plugins to start working this way as the configuration confused a lot of people.

Reply:
To be clear this changed we should check if `options` is undefined or has devtool. If it's defined but doesn't have devtool we should throw (that means the user is using the old usage, so we could warn them)

Reply:
@timneutkens  what the purpose of refactoring it that way?

Reply:
A few reasons:

1. Like I said above: The configuration of multiple plugins together confused a lot of people.
2. This avoids conflicts in naming, what if some other plugin also adds a `devtool` option

Reply:
:+1:

Reply:
@timneutkens what about save compatibility between versions?
```
module.exports = (options = {}) => {
  const configurator = (nextConfig = {}) => {
    ...
  }

  if (options.devtool) {
    return configurator;
  }

  return configurator(options);
} 
```

Reply:
I'd rather make sure everyone uses the correct syntax, but this seems alright, all we have to do then is `console.warn` to use the new syntax 👍